<<<<<<< HEAD:Waterfall-Proxy-Patches/0013-Change-state-to-encrypt-only-if-onlineMode-is-true.patch
From 9ae6630c64fdf378d314b427ccbd11e77c969ed8 Mon Sep 17 00:00:00 2001
=======
From bb419bd35b5e513be0ca20cda4c97bcaa23f0cd7 Mon Sep 17 00:00:00 2001
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0012-Change-state-to-encrypt-only-if-onlineMode-is-true.patch
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Wed, 16 Dec 2020 18:25:03 +0800
Subject: [PATCH] Change state to encrypt only if onlineMode is true


diff --git a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
<<<<<<< HEAD:Waterfall-Proxy-Patches/0013-Change-state-to-encrypt-only-if-onlineMode-is-true.patch
index 92c97ac0..9293739f 100644
=======
index 37784acd..1811084a 100644
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0012-Change-state-to-encrypt-only-if-onlineMode-is-true.patch
--- a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
@@ -426,11 +426,12 @@ public class InitialHandler extends PacketHandler implements PendingConnection
                 if ( onlineMode )
                 {
                     unsafe().sendPacket( request = EncryptionUtil.encryptRequest() );
+                    // FlameCord - Change state to encrypt only if onlineMode is true
+                    thisState = State.ENCRYPT;
                 } else
                 {
                     finish();
                 }
-                thisState = State.ENCRYPT;
             }
         };
 
-- 
2.32.0.windows.1

